src/services/housing.js

```diff
@@ -0,0 +1,208 @@
+ * @property {Number} Points The number of application points for this student (only visible to housing admin)
+ */
+
+/**
+ * Note: Properties 'HallRank' and 'HallName' must be capitalized to match the backend
+ * @global
+ * @typedef ApartmentChoice
+ * @property {Number} [ApplicationID] Application ID number of this application
+ * @property {Number} HallRank The rank assigned to this hall by the user
+ * @property {String} HallName The name of the apartment hall
+ */
+
+/**
+ * @global
+ * @typedef ApplicationDetails
+ * @property {Number} ApplicationID Application ID number of this application
+ * @property {*} DateSubmitted The date the application was submitted, or null if not yet submitted
+ * @property {*} DateModified The date the application was last modified
+ * @property {String} EditorUsername Username of the application editor
+ * @property {String} Gender Gender
+ * @property {FullApplicantInfo[]} Applicants Array of student usernames
+ * @property {ApartmentChoice[]} ApartmentChoices Array of ApartmentChoice objects
+ * @property {Number} TotalPoints The total application points associated with this application
+ * @property {Number} AvgPoints The average application points per applicant
+ */
+
+/**
+ * Check if the current user is authorized to view the housing staff page for applications
+ * @return {Promise.<Boolean>} True if the user is authorized to view the housing application staff page
+ */
+const checkHousingAdmin = async () => {
+  try {
+    return await http.get(`housing/admin`);
+  } catch {
+    return false;
+  }
+};
+
+/**
+ * Add a user to the housing admin whitelist
+ * @param {String} [username] Username in firstname.lastname format
+ * @return {Response} response of http request
+ */
+const addHousingAdmin = (username) => {
+  return http.post(`housing/admin/${username}/`);
+};
+
+/**
+ * Delete a user to the housing admin whitelist
+ * @param {String} [username] Username in firstname.lastname format
+ * @return {Response} response of http request
+ */
+const deleteHousingAdmin = (username) => {
+  return http.del(`housing/admin/${username}/`);
+};
+
+/**
+ * Check if a given student is on an existing application from the current semester
+ * @param {String} [username] Username in firstname.lastname format
+ * @return {Promise.<Number>} Application's ID number
+ */
+const getCurrentApplicationID = async (username) => {
+  let applicationID;
+  let err;
+  try {
+    if (username) {
+      applicationID = await http.get(`housing/apartment/${username}/`);
+    } else {
+      applicationID = await http.get('housing/apartment');
+    }
+  } catch (e) {
+    // handle thrown 404 errors
+    if (e.status !== 404) throw e;
+    err = e;
+    applicationID = false;
+    console.log('A 404 code indicates that an application was not found for this applicant');
+  }
+  // handle "unhandled" requests, `this.status = 404`s, and 404 errors
+  var status = (this.status = (err && err.status) || this.status || 404);
```

Comment:
I'm not sure this code is correct, and if it is, it could probably be cleaned up. Firstly, in an arrow function like this, the `this` keyword is lexically bound which in this case means it will refer to the `window` object. I don't think you mean to be reading (and setting !) the `status` property on the window object (here's some detail on [what `window.status` is](https://developer.mozilla.org/en-US/docs/Web/API/Window/status)). It seems like you're trying to set the variable defined by `var status` to the value of `err.status`, or `404` if there is no `err.status`. I think `const status = err?.status ?? 404` would suffice.

Also, please avoid using the `var` keyword to define variables. It has unexpected scoping behaviors which can result in hard-to-notice bugs. Use `const` for a variable like this that doesn't need to be mutated after it's defined, or `let` for a mutable value.

Reply:
Yeah, you can tell I didn't check it very hard that night just by the fact that there is a `var` in it. As someone whose favorite language is C++, I despise the behavior of `var` 😆